tests/test_package_manager/test_v1.py

```diff
@@ -0,0 +1,234 @@
+)
+
+
+class TestPackageManagerV1(BaseAEATestCase):
+    """Test aea package manager."""
+
+    use_packages_dir: bool = True
+    packages_json_file: Path
+
+    @classmethod
+    def setup_class(cls) -> None:
+        """Setup class."""
+
+        super().setup_class()
+        cls.packages_json_file = cls.t / PACKAGES / "packages.json"
+
+    def test_initialization(
+        self,
+    ) -> None:
+        """Test object initialization."""
+
+        pm = PackageManagerV1.from_dir(self.packages_dir_path)
+        packages = json.loads(self.packages_json_file.read_text())
+
+        assert len(pm.dev_packages) == len(packages["dev"])
+        assert len(pm.third_party_packages) == len(packages["third_party"])
+
+        assert pm.path == self.packages_dir_path
+        assert pm.get_package_hash(package_id=EXAMPLE_PACKAGE_ID) is not None
+        assert pm.get_package_hash(package_id=DUMMY_PACKAGE_ID) is None
+
+    def test_sync(
+        self,
+    ) -> None:
+        """Test sync."""
+
+        pm = PackageManagerV1(
+            path=self.packages_dir_path,
+            dev_packages=OrderedDict({DUMMY_PACKAGE_ID: DUMMY_PACKAGE_HASH}),
+        )
+
+        with pytest.raises(
+            ValueError,
+            match=re.escape(
+                "Both `update_packages` and `update_hashes` cannot be set to `True`."
+            ),
+        ):
+            pm.sync(update_hashes=True, update_packages=True)
+
+        with mock.patch.object(pm, "add_package") as update_patch:
+            pm.sync(dev=True, third_party=False)
+            update_patch.assert_called_with(package_id=DUMMY_PACKAGE_ID)
+
+        pm = PackageManagerV1(
+            path=self.packages_dir_path,
+            third_party_packages=OrderedDict({DUMMY_PACKAGE_ID: DUMMY_PACKAGE_HASH}),
+        )
+
+        with mock.patch.object(pm, "add_package") as update_patch:
+            pm.sync(dev=False, third_party=True)
+            update_patch.assert_called_with(package_id=DUMMY_PACKAGE_ID)
+
+
+class TestHashUpdateDev(BaseAEATestCase):
+    """Test hash update."""
+
+    use_packages_dir: bool = True
+
+    def test_update_package_hashes(
+        self,
+    ) -> None:
+        """Test update package hash method."""
+
+        packages_v1 = json.loads(PACKAGE_JSON_FILE.read_text(encoding="utf-8"))
+
+        original_hash = packages_v1["dev"][EXAMPLE_PACKAGE_ID.to_uri_path]
+        packages_v1["dev"][EXAMPLE_PACKAGE_ID.to_uri_path] = DUMMY_PACKAGE_HASH
+
+        packages_json_file = self.packages_dir_path / "packages.json"
+        packages_json_file.write_text(json.dumps(obj=packages_v1))
+        pm = PackageManagerV1.from_dir(self.packages_dir_path)
+
+        pm.update_package_hashes().dump()
+
+        packages_v1_updated = json.loads(packages_json_file.read_text(encoding="utf-8"))
+
+        assert pm.dev_packages[EXAMPLE_PACKAGE_ID] == original_hash
+        assert (
+            packages_v1_updated["dev"][EXAMPLE_PACKAGE_ID.to_uri_path] == original_hash
+        )
```

Comment:
this code is an almost exact copy of the `v0` version where it's called `TestHashUpdate`. This makes it an unnecessarily painful exercise for the reviewer; it could have been a parametrized test. 
https://github.com/valory-xyz/open-aea/blob/46cbcc4e6eebc15947e418e75e1d493c3c3e8eac/tests/test_package_manager/test_v0.py#L85-L110